### PR TITLE
[T896][ADD] website_blog_mail_server

### PR DESCRIPTION
**Pull request by TimLai125**
Task #[896](https://www.quartile.co/web?debug=#id=896&action=771&model=project.task&view_type=form&menu_id=505)

- Refactor `website_forum_adj`: Put the logic of changing the `email_from` of the email into `ir_mail_server_email_from`

### CODE REVIEW

File: website_blog_mail_server/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2020 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+{
+    'name': 'Website blog\'s Mail Server',
```
Comment (yostashiro) [inline website_blog_mail_server/__manifest__.py]:
```suggestion
    'name': 'Website Blog Mail Server',
```

File: website_blog_mail_server/models/mail_message.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class MailMessage(models.Model):
+    _inherit = 'mail.message'
+
+    @api.model
+    def _get_reply_to(self, values):
+        res = super(MailMessage, self)._get_reply_to(values)
+        if 'model' in values:
+            print(values['model'])
```
Comment (yostashiro) [inline website_blog_mail_server/models/mail_message.py]: ?

File: website_blog_mail_server/models/website.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2020 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import fields, models
+
+
+class Website(models.Model):
+    _inherit = 'website'
+
+    website_blog_mail_server_id = fields.Many2one(
```
Comment (yostashiro) [inline website_blog_mail_server/models/website.py]:
```suggestion
    blog_mail_server_id = fields.Many2one(
```

I prefer field names to be concise.

File: website_blog_mail_server/models/res_partner.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    @api.model
+    def _notify_prepare_email_values(self, message):
+        mail_values = super(
+            ResPartner, self)._notify_prepare_email_values(message)
+        if message.model and message.model in ('blog.post', 'blog.blog'):
+            website_blog_mail_server_id = self.env['website'].sudo(
+            ).get_current_website().website_blog_mail_server_id
+            if website_blog_mail_server_id:
+                mail_values['mail_server_id'] = website_blog_mail_server_id.id
```
Comment (yostashiro) [inline website_blog_mail_server/models/res_partner.py]:
```suggestion
            blog_mail_server_id = self.env['website'].sudo(
            ).get_current_website().blog_mail_server_id
            if blog_mail_server_id:
                mail_values['mail_server_id'] = blog_mail_server_id.id
```

Just to make the variable names shorter.

File: website_blog_mail_server/models/mail_message.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+        if 'model' in values and values['model'] in ('blog.blog', 'blog.post'):
+            website_blog_mail_server_id = self.env['website'].sudo().get_current_website().website_blog_mail_server_id
+            if website_blog_mail_server_id and website_blog_mail_server_id.smtp_from:
+                return "<%s>" % website_blog_mail_server_id.smtp_from
```
Comment (yostashiro) [inline website_blog_mail_server/models/mail_message.py]:
```suggestion
            blog_mail_server_id = self.env['website'].sudo().get_current_website().blog_mail_server_id
            if blog_mail_server_id and blog_mail_server_id.smtp_from:
                return "<%s>" % blog_mail_server_id.smtp_from
```

Just to make the variable names shorter.

File: website_blog_mail_server/models/res_config_settings.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+from odoo import models, fields
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = 'res.config.settings'
+
+    website_blog_mail_server_id = fields.Many2one(
+        'ir.mail_server',
+        related='website_id.website_blog_mail_server_id',
```
Comment (yostashiro) [inline website_blog_mail_server/models/res_config_settings.py]:
```suggestion
    blog_mail_server_id = fields.Many2one(
        'ir.mail_server',
        related='website_id.blog_mail_server_id',
```

**Review by yostashiro — CHANGES_REQUESTED** [review]
Can you please change the module name from `ir_mail_server_email_from` to `ir_mail_server_custom_smtp` or something like that. Just thought "email_from" could be a bit misleading (one may think about the sender email address).

**Comment by TimLai125** [comment]
> Can you please change the module name from `ir_mail_server_email_from` to `ir_mail_server_custom_smtp` or something like that. Just thought "email_from" could be a bit misleading (one may think about the sender email address).

I will keep `ir_mail_server_email_from` as this module is used to overwrite the sender email address.

**Comment by TimLai125** [comment]
Closed the PR for now, we will backport the mail_outbound_static OCA module to v11.0.

**Pull request closed without merging**